src/infill/ZigzagConnectorProcessorNoEndPieces.cpp

```diff
@@ -33,13 +33,20 @@ void ZigzagConnectorProcessorNoEndPieces::registerScanlineSegmentIntersection(co
     else
     {
         if (previous_scanline_is_even && !this_scanline_is_even)
-        { // add whole zigzag_connector (including the just obtained point)
-            for (unsigned int point_idx = 1; point_idx < zigzag_connector.size(); point_idx++)
+        {
+            if (skip_some_zags && ++current_zag_count >= zag_skip_count)
             {
-                addLine(zigzag_connector[point_idx - 1], zigzag_connector[point_idx]);
+                // skip this zag and reset the count
+                current_zag_count = 0;
+            }
+            else
+            {   // add whole zigzag_connector (including the just obtained point)
+                for (unsigned int point_idx = 1; point_idx < zigzag_connector.size(); point_idx++)
+                {
+                    addLine(zigzag_connector[point_idx - 1], zigzag_connector[point_idx]);
+                }
+                addLine(zigzag_connector.back(), intersection);
             }
-            addLine(zigzag_connector.back(), intersection);
-            zigzag_connector.clear();
```

Comment:
Outside this big if block there is a `zigzag_connector.clear();` which will always be called, so this is kinda redundant.